packages/sanes-chrome-extension/src/extension/background/model/persona/index.unit.spec.ts

```diff
@@ -107,6 +110,91 @@ withChainsDescribe('Persona', () => {
+
+      {
+        const connection = await BnsConnection.establish(bnsUrl);
+        const profile = new UserProfile();
+        const wallet = profile.addWallet(Ed25519HdWallet.fromMnemonic(mnemonic));
+        const identity0 = await profile.createIdentity(wallet.id, connection.chainId(), HdPaths.iov(0));
+        const identity1 = await profile.createIdentity(wallet.id, connection.chainId(), HdPaths.iov(1));
+        const identity2 = await profile.createIdentity(wallet.id, connection.chainId(), HdPaths.iov(2));
+
+        const registerName0: RegisterUsernameTx = {
+          kind: 'bns/register_username',
+          creator: identity0,
+          addresses: [],
+          username: name0,
+        };
+
+        const registerName1: RegisterUsernameTx = {
+          kind: 'bns/register_username',
+          creator: identity1,
+          addresses: [],
+          username: name1,
+        };
+
+        const registerName2a: RegisterUsernameTx = {
+          kind: 'bns/register_username',
+          creator: identity2,
+          addresses: [],
+          username: name2[0],
+        };
+
+        const registerName2b: RegisterUsernameTx = {
+          kind: 'bns/register_username',
+          creator: identity2,
+          addresses: [],
+          username: name2[1],
+        };
+
+        const transactions = [[registerName0], [registerName1], [registerName2a, registerName2b]];
+        const nonces = [
+          await connection.getNonces({ pubkey: identity0.pubkey }, 1),
+          await connection.getNonces({ pubkey: identity1.pubkey }, 1),
+          await connection.getNonces({ pubkey: identity2.pubkey }, 2),
+        ];
+
+        for (const creatorIndex of [0, 1, 2]) {
+          for (let i = 0; i < transactions[creatorIndex].length; ++i) {
+            const tx = transactions[creatorIndex][i];
+            const nonce = nonces[creatorIndex][i];
+
+            const fee = await connection.getFeeQuote(tx);
+            const signed = await profile.signTransaction({ ...tx, fee: fee }, bnsCodec, nonce);
+            await connection.postTx(bnsCodec.bytesToPost(signed));
+          }
+        }
+        connection.disconnect();
+      }
+
+      // Due to missing WebSocket connection, we cannot subscribe to block. Wait instead.
+      await sleep(2000);
+
+      const db = new Db().getDb();
+      const signingServer = new SigningServer();
+      const persona = await Persona.create(db, signingServer, 'passwd', mnemonic);
+      await persona.createAccount(db); // index 1
+      await persona.createAccount(db); // index 2
+      await persona.createAccount(db); // index 3
+
+      const accounts = await persona.getAccounts();
+      expect(accounts.length).toEqual(4);
+      expect(accounts[0].label).toEqual(`${name0}*iov`);
+      expect(accounts[1].label).toEqual(`${name1}*iov`);
+      expect(accounts[2].label).toEqual('Multiple names');
+      expect(accounts[3].label).toEqual('Account 3');
```

Comment:
I was thinking that maybe we should increase the counter, starting from 0 and having some accounts with name, then see "Account 3"  but being the fourth one, could be misleading.

Thoughts?

Reply:
Yeah, right, I don't know which one is better. On the one hand we have the natural counting account 1, 2, 3, …. On the other hand each account belings to one derication index 0, 1, 2, 3.

So far we started with "Account 0", but this can easily be changed. I have no preference.

Reply:
Me too, no preference.

Reply:
Let the users decide …